src/lib/math.ts

```diff
@@ -0,0 +1,83 @@
+export const MATHJAX_URL = 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js?delayStartupUntil=configured'
+
+const configureMathJax = (): void => {
+  MathJax.Hub.Config({
+    config: ['Accessible.js', 'Safe.js'],
+    errorSettings: { message: ['!'] },
+    skipStartupTypeset: true,
+    messageStyle: 'none'
+  })
+  window.MathJax.Hub.processSectionDelay = 0
+  MathJax.Hub.Configured()
+}
+
+const loadMathJaxScript = async (): Promise<void> => {
+  const head = document.getElementsByTagName('head')[0]
+  const maybeMathJaxScript = head.querySelector(`script[src="${MATHJAX_URL}"]`)
+  let loadedPromise: Promise<void> | null = null
+
+  // Check if the script is already populated
+  if (maybeMathJaxScript === null) {
+    const script = document.createElement('script')
+    script.type = 'text/javascript'
+    script.src = MATHJAX_URL
+    head.appendChild(script)
+
+    loadedPromise = new Promise<void>((resolve) => {
+      const configureCallback = (): void => {
+        configureMathJax()
+        resolve()
+      }
+      script.onload = configureCallback
+    })
+  } else {
+    // Chain callbacks so we have something to await on
+    const script = maybeMathJaxScript as HTMLScriptElement
+    const oldOnloadCallback = script.onload as () => void
+
+    loadedPromise = new Promise<void>((resolve) => {
+      const chainedCallback = (): void => {
+        if (oldOnloadCallback !== null) {
+          oldOnloadCallback()
+        } else {
+          configureMathJax()
+        }
+        resolve()
+      }
+
+      script.onload = chainedCallback
+    })
+  }
```

Comment:
I probably need some clarification on what this condition is doing with chained call backs. How does this affect the manner in which mathjax is loaded?